src/rgw/rgw_admin.cc

```diff
@@ -9360,6 +9374,43 @@ int main(int argc, const char **argv)
     }
   }
 
+  // TODO: remove a module from allowlist
+
+  if (opt_cmd == OPT::SCRIPT_MODULE_ADD) {
+#ifdef WITH_RADOSGW_LUA_MODULES
+    if (!script_module) {
+      cerr << "ERROR: lua module name was not provided (via --module)" << std::endl;
+      return EINVAL;
+    }
+    const auto rc = rgw::lua::add_module(store, null_yield, *script_module, bool(allow_compilation));
+    if (rc < 0) {
+      cerr << "ERROR: failed to add lua module: " << script_module << " .error: " << rc << std::endl;
+      return rc;
+    }
+#else
+    cerr << "ERROR: adding lua modules in not permitted" << std::endl;
+#endif
+  }
+
+  if (opt_cmd == OPT::SCRIPT_MODULE_LIST) {
+#ifdef WITH_RADOSGW_LUA_MODULES
+    rgw::lua::modules_t modules;
+    const auto rc = rgw::lua::list_modules(store, null_yield, modules);
+    if (rc == -ENOENT) {
+      std::cout << "no lua modules in allowlist" << std::endl;
+    } else if (rc < 0) {
+      cerr << "ERROR: failed to read lua modules allowlist. error: " << rc << std::endl;
+      return rc;
+    } else {
+      std::for_each(modules.begin(), modules.end(), [](const std::string& module_str) {
+          std::cout << module_str << std::endl;
+      });
+    }
+#else
+    cerr << "ERROR: adding lua modules in not permitted" << std::endl;
```

Comment:
s/adding/listing/

why not just bail out at seeing unknown options?

Reply:
> s/adding/listing/
> 
+1

> why not just bail out at seeing unknown options?

it will require ifdefing in lots of places